volatility3/framework/symbols/linux/extensions/__init__.py

```diff
@@ -307,6 +307,22 @@ def section_strtab(self):
 
 
 class task_struct(generic.GenericIntelProcess):
+    def has_valid_name_and_pid(self) -> bool:
+        """
+        Ensures the pid and name for this process are sane
+        When these are broken it is a sure sign of smear
+        during process enumeration
+        """
+        try:
+            pid = self.pid
+            name = utility.array_to_string(self.comm)
+        except exceptions.InvalidAddressException:
+            return False
+
+        # ensure the pid within the bounds of a signed int
+        # and that we have something of a name
+        return (0 < pid < 2147483647) and len(name) > 0
```

Comment:
If the type of pid is signed int, can it ever be outside these bounds?

Reply:
Aren't we opening the door to task hiding in memory forensics with this? Writing \x00 to the comm field could even be done from user space with just a syscall. Also, we definitely shouldn't use `break` there. Doing so not only allows a single task to hide but also exposes all subsequent tasks to the same risk.

Reply:
Minor but technically speaking it has to be  (0 < pid <= 2147483647) 
```python
>>> struct.unpack('i', b"\xff\xff\xff\x7f")
(2147483647,)
>>> struct.unpack('i', b"\xff\xff\xff\x80")
(-2130706433,)
```